rules/docker-tam.mk

```diff
@@ -0,0 +1,29 @@
+# docker image for TAM
+
+DOCKER_TAM_STEM = docker-tam
+DOCKER_TAM = $(DOCKER_TAM_STEM).gz
+DOCKER_TAM_DBG = $(DOCKER_TAM_STEM)-$(DBG_IMAGE_MARK).gz
+
+$(DOCKER_TAM)_PATH = $(DOCKERS_PATH)/$(DOCKER_TAM_STEM)
+
+$(DOCKER_TAM)_DEPENDS += $(SONIC_TAM) $(PROTOBUF)
+$(DOCKER_TAM)_DBG_DEPENDS = $($(DOCKER_CONFIG_ENGINE_STRETCH)_DBG_DEPENDS)
+$(DOCKER_TAM)_DBG_DEPENDS += $(SONIC_TAM) $(PROTOBUF)
+$(DOCKER_TAM)_DBG_IMAGE_PACKAGES = $($(DOCKER_CONFIG_ENGINE_STRETCH)_DBG_IMAGE_PACKAGES)
+
+SONIC_DOCKER_IMAGES += $(DOCKER_TAM)
+SONIC_STRETCH_DOCKERS += $(DOCKER_TAM)
+SONIC_INSTALL_DOCKER_IMAGES += $(DOCKER_TAM)
```

Comment:
This container is for a very specific use case and should be optionally included into SONiC image.
Making it included by default increases image size and build time since it is the only dependency for protobuf.

Reply:
HI Marian,

I've checked and we can make the change to not include TAM docker by default. I'll make the changes accordingly and update the diffs.

Thanks & Regards,
Shirisha.

Reply:
Hi Marian,

While analyzing this further,  we realized this may not be a clean approach since this only turns the container off. Other parts of the feature functionality like CLI, Orch for this feature will still be built in the image and could be exercised with incomplete functionality.
 
I would like to keep this feature intact for the time being. We could turn the feature functionality off end-to-end through a PR if required.

Reply:
Hi Marian, 

We also tried to use the capability check introduced in SAI via PR https://github.com/opencomputeproject/SAI/pull/989 to disable the feature functionality at Orch and CLI by default. However, this PR support is missing on Azure:master since SAI version used is 1.5. 

We can handle this change too when the SAI API version gets advanced.

Reply:
Hi Marian,

I have implemented the changes required to include the TAM container optionally into SONiC build. It is turned off by default. Have uploaded the latest diffs. Please review and approve the changes.

Regards,
Shirisha.